docs/getting-started/github-apps.rst

```diff
@@ -45,47 +44,117 @@ sync with the linked GitHub repository.
+along with explanations for each field. For every branch/release on GitHub that has one of these files, a corresponding entry
 will be made on Dockstore.
 
 Workflow YML File
 ++++++++++++++++++
+For a workflow, the ``/.dockstore.yml`` has the following general structure
 
 .. code:: yaml
 
    version: 1.2
    workflows:
-      - name: aligner
-        subclass: CWL
-        primaryDescriptorPath: /Dockstore.cwl
-        testParameterFiles:
-            - /test/dockstore.cwl.json
+      - name: <String>
+        subclass: <CWL | WDL | NFL | GXFORMAT2>
+        primaryDescriptorPath: <String>
+        testParameterFiles: <String Array>
 
 version
     The version of the .dockstore.yml schema. Currently at 1.2.
 workflows
     An array of workflows. Each element corresponds to a workflow on Dockstore.
-name
+name (optional)
     The optional workflow name that is used to uniquely identify workflows in repositories with multiple workflows.
+    **Each workflow listed must have a unique (or no) name.**
 subclass
-    The descriptor language used for the workflow. Supported values include CWL, WDL, and NFL.
+    The descriptor language used for the workflow. Supported values include CWL, WDL, NFL (Nextflow), and GXFORMAT2 (Galaxy).
 primaryDescriptorPath
-    The path to the primary descriptor file in the Git repository
-testParameterFiles
-    An array of paths to test parameter files in the Git repository.
+    The absolute path to the primary descriptor file in the Git repository
+testParameterFiles (optional)
+    An array of absolute paths to test parameter files in the Git repository.
+
+Ex. .dockstore.yml with a single workflow
+
+.. code:: yaml
+
+   version: 1.2
+   workflows:
+      - subclass: CWL
+        primaryDescriptorPath: /Dockstore.cwl
+        testParameterFiles:
+            - /test/dockstore.cwl.json
+
+The above ``.dockstore.yml`` is for a single workflow. Note that the name is not present since it is optional.
+
+Ex. .dockstore.yml with multiple workflows
+
+.. important:: The **name** field is an optional field used when a repository has multiple workflows in it that a user wants to register
+    as separate entries on Dockstore. Each entry within a ``.dockstore.yml`` file corresponds to a unique entry on Dockstore.
+
+.. code:: yaml
+
+   version: 1.2
+   workflows:
+      - name: globalAligner
+        subclass: CWL
+        primaryDescriptorPath: /runGlobalAligner.cwl
+        testParameterFiles:
+            - /test/globalAligner.cwl.json
+      - name: localAligner
+        subclass: CWL
+        primaryDescriptorPath: /runLocalAligner.cwl
+        testParameterFiles:
+            - /test/localAligner.cwl.json
+
+A common pattern seen on Dockstore is GitHub repositories that store many workflows. The above ``.dockstore.yml``
+has two entries for workflows. Notice that each entry uses a different name. Names are required if you want 
```

Comment:
Out of curiosity is requiring the name for multiple workflows in the same repo a departure from manual register? This is the sam as 'extra' name right?

Reply:
This is the same as the extra workflow name, which is required if you have multiple workflows from the same repository. One of these workflows could have no name and it would work, as long as two of them didn't have the same name.